src/main/java/fr/insee/pearljam/batch/service/impl/CommunicationServiceImpl.java

```diff
@@ -66,20 +67,22 @@ public BatchErrorCode handleCommunications() throws SynchronizationException, Mi
                 surveyUnitDao.getSurveyUnitsById(surveyUnitIds).stream()
                         .collect(Collectors.toMap(SurveyUnitType::getId, su -> su));
 
-        // retrieve all matching template IDs
-        List<Long> templateIds = communicationsToSend.stream()
-                .map(CommunicationRequestType::getCommunicationTemplateId)
-                .distinct().toList();
+// retrieve all matching campaignId and meshuggahId pairs
+        List<Pair<String, String>> campaignMeshuggahPairs = communicationsToSend.stream()
+            .map(communication -> Pair.of(communication.getCampaignId(), communication.getMeshuggahId()))
```

Comment:
suggestion : if CommunicationTemplateDbID class exist in batch, it could replace the usage of Pair class, which left and right methods could be messed with
Collection to Map<String,String> could be an option here